Build fails with Visual Studio 2012

As of 20/1/2016, using the Visual Studio 2012 (xp toolset), SpvBuilder fails - this may be due to unsupported C++11 or C++14 features:

    glslang\SPIRV\SpvBuilder.cpp(1807): error C2143: syntax error : missing ')' before '{'         [glslang\build\x64\SPIRV\SPIRV.vcxproj]
    glslang\SPIRV\SpvBuilder.cpp(1807): error C2661: 'std::stack<_Ty>::push' : no overloaded function takes 0 arguments [glslang\build\x64\SPIRV\SPIRV.vcxproj]
    glslang\SPIRV\SpvBuilder.cpp(1807): error C2143: syntax error : missing ';' before '{' [glslang\build\x64\SPIRV\SPIRV.vcxproj]
    glslang\SPIRV\SpvBuilder.cpp(1807): error C2143: syntax error : missing ';' before '}' [glslang\build\x64\SPIRV\SPIRV.vcxproj]
    glslang\SPIRV\SpvBuilder.cpp(1807): error C2059: syntax error : ')' [glslang\build\x64\SPIRV\SPIRV.vcxproj]

--- a/SPIRV/SpvBuilder.cpp
+++ b/SPIRV/SpvBuilder.cpp
@@ -1804,7 +1804,9 @@ Block& Builder::makeNewBlock()
 
 Builder::LoopBlocks& Builder::makeNewLoop()
 {
-    loops.push({makeNewBlock(), makeNewBlock(), makeNewBlock(), makeNewBlock()});
+    // Older MSVC versions don't allow inlining of blocks below.
+    LoopBlocks blocks = {makeNewBlock(), makeNewBlock(), makeNewBlock(), makeNewBlock()};
+    loops.push(blocks);
     return loops.top();
 }
 

